react-sdk/src/providers/tambo-registry-provider.tsx

```diff
@@ -178,6 +195,14 @@ export const TamboRegistryProvider: React.FC<
     }
   }, [registerTools, userTools]);
 
+  useEffect(() => {
+    if (mcpServers) {
+      fetchToolsFromMCPServers(mcpServers).then((tools) => {
```

Comment:
Use async here rather than then() - it's harder to maintain 

(Use an inner async function that you call from inside useEffect)

Reply:
This can also get you cancellation via an abortcontroller